### PR TITLE
Add shortcut action to add AuxKernel nested within an AuxVariable

### LINKED ISSUE #20753: Shortcut AuxKernel action
## Reason

Right now we can do
```python
[AuxVariables]
  [foo]
    order = SECOND
    [InitialCondition]
      type = bar
    []
  []
[]
```
but unfortunately not
```python
[AuxVariables]
  [foo]
    order = SECOND
    [AuxKernel]
      type = bar
    []
  []
[]
```
I don't see why we aren't allowed to do so...

## Design

Add an action and associate it to the proper syntax.

## Impact

More concise input files.

### PR DESCRIPTION
**Pull request by hugary1995**
close #20753

This PR allows us to do
```python
[AuxVariables]
  [foo]
    order = SECOND
    [AuxKernel]
      type = bar
    []
  []
[]
```

### CODE REVIEW

**Comment by dschwen** [comment]
I like this.

**Comment by hugary1995** [comment]
You live in github! :rofl:

**Comment by YaqiWang** [comment]
There could be multiple aux kernels for the same aux var. But I like this syntax too.

**Comment by hugary1995** [comment]
That's a good point. I could imagine cases where multiple auxkernels for the same aux var with different execute ons or different block/boundary restrictions. The proposed syntax in this PR should be able handle those cases properly.

Looking at the current SQA, I don't see any mentions of "The framework shall support multiple boundary-restricted auxkernels for the same auxvariable..." or anything like that. That's probably a separate issue/PR.

File: framework/include/actions/AddAuxKernelAction.h [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+#include "MooseObjectAction.h"
+
```
Comment (GiudGiud) [inline framework/include/actions/AddAuxKernelAction.h:13]: docstring

File: test/tests/auxkernels/array_aux_kernels/function_array_aux.i [hunk 2]
```diff
@@ -58,11 +62,6 @@
     variable = y
     functions = 'func1 func2'
   []
-  [z]
-    type = FunctionArrayAux
-    variable = z
-    functions = 'func1 func2 func3'
-  []
```
Comment (GiudGiud) [inline test/tests/auxkernels/array_aux_kernels/function_array_aux.i:65]: duplicate the test and make sure they compare the same
use active = '' / inactive='' to have both syntaxes in the file (and avoid duplicating the file)

**Review by GiudGiud — CHANGES_REQUESTED** [review]
Missing a short doc page for the action that shows the new syntax.
Tests configuration file should have the new issue and new design file attached to a new test

**Review by GiudGiud — DISMISSED** [review]

**Comment by GiudGiud** [comment]
it's a reading/writing exodus with hdf5 issue. 
3 options:
- revert to an old conda to generate the file
- use an old exodus file from another test
- wait for the libmesh submodule update to be merged in, rebase, regenerate your file

Tagging @cticenhour for awareness.

**Review by GiudGiud — APPROVED** [review]

**Pull request merged**